src/Polykey.ts

```diff
@@ -394,22 +358,33 @@ export default class Polykey {
   }
 
   initSync(): void {
+    // Initialize peer store and peer discovery classes
     // check if .polykey exists
     //  make folder if doesn't
     if (!this.fs.existsSync(this.polykeyPath)) {
       this.fs.mkdirSync(this.polykeyPath, {recursive: true})
+    } else if (this.fs.existsSync(this.metadataPath)) {
+      const fileContents = jsonfile.readFileSync(this.metadataPath)
+      this.metadata = {
+        vaults: fileContents.vaults,
+        peerInfo: fileContents.peerInfo,
+        publicKeyPath: fileContents.publicKeyPath,
+        privateKeyPath: fileContents.privateKeyPath
+      }
+    } else {
+      // Create a new peerInfo
+      const peerInfo = new PeerInfo(this.keyManager.getPublicKey())
       const metadataTemplate = {
         vaults: {},
-        publicKeyPath: null,
-        privateKeyPath: null,
-        passphrase: null
+        peerInfo: RPCMessage.encodePeerInfo(peerInfo)
       }
       jsonfile.writeFileSync(this.metadataPath, metadataTemplate)
       this.metadata = metadataTemplate
-    } else if (this.fs.existsSync(this.metadataPath)) {
-      this.metadata = jsonfile.readFileSync(this.metadataPath)
     }
 
+    this.peerStore = new PeerStore(RPCMessage.decodePeerInfo(this.metadata.peerInfo))
+    this.peerDiscovery = new PeerDiscovery(this.peerStore, this.keyManager)
```

Comment:
Similar issue to the git PR, I think the `Polykey` class is doing too many things. We can make it more cohesive.

Reply:
Well there are some things that we can remove from the `Polykey` class, the vault operations for example (e.g. secret opperations

Reply:
Well there are some things that we can remove from the `Polykey` class, the vault operations for example (e.g. secret operations). It makes sense that the `Polykey` class should only be responsible for the highest order operations like creating/destroying vaults. I'm not sure how we would separate out peer discovery though, perhaps we could make it a separate class like the `KeyManager`?

Reply:
I have removed vault operations from Polykey to reduce its domain model scope. Now one has to `getVault` and then can do things like `vault.addSecret(...)`